packages/components/table/src/table-header/event-helper.ts

```diff
@@ -160,8 +160,22 @@ function useEvent<T>(props: TableHeaderProps<T>, emit) {
     event.stopPropagation()
     const order =
       column.order === givenOrder ? null : givenOrder || toggleOrder(column)
-
     const target = (event.target as HTMLElement)?.closest('th')
+    const clickTarget = event.currentTarget
+
+    if (
+      hasClass(clickTarget, 'ascending') &&
+      !column.sortOrders.includes('ascending')
+    ) {
+      return
+    }
+
+    if (
+      hasClass(clickTarget, 'descending') &&
+      !column.sortOrders.includes('descending')
+    ) {
+      return
+    }
```

Comment:
Wouldn't it be more appropriate to put this part of the code after `if (!column.sortable) return`?

Reply:
Thanks for the correction, changes have been made